internal/confmapprovider/discovery/discoverer.go

```diff
@@ -449,11 +460,15 @@ func (d *discoverer) updateReceiverForObserver(receiverID component.ID, receiver
 
 func factoryForObserverType(extType component.Type) (otelcolextension.Factory, error) {
 	factories := map[component.Type]otelcolextension.Factory{
-		component.MustNewType("docker_observer"):   dockerobserver.NewFactory(),
```

Comment:
Just curious why are we removing it?

Reply:
It is not being added by default on Windows, this is temporary until next OTel deps upgrade, since then we will have the docker observer fixed for Windows.

Reply:
Oh, I forgot the context! These changes are only for Windows binary, correct?

Reply:
Then it is fine. I can go ahead with LGTM